### PR TITLE
Add adress info Component

### LINKED ISSUE #23: Create component to display address info
My two cents:
I'm thinking about making all the red areas optional and dividing green areas into different components.

That is, a component for the address and `copy`, `open on etherscan` buttons (as we can use it for tx hashes, not only for addresses)
Then the final component that we see here.

@alongoni, @nicosampler

### PR DESCRIPTION
**Pull request by alongoni**
Closes #23.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

File: src/navigation/EllipsisMenu/index.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+};
+
+type Props = {
+  menuItems: EllipsisMenuItem[];
+};
+
+const EllipsisMenu = ({ menuItems }: Props): React.ReactElement => {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+
+  const handleClick = (event: any) => setAnchorEl(event.currentTarget);
```
Comment (nicosampler) [inline src/navigation/EllipsisMenu/index.tsx:54]: I could not add the type, perhaps someone could help me with it.

**Comment by nicosampler** [comment]
once CopyToClipboardBtn component is merged, I need to replace the fixed icon "copy" by the component.

File: src/navigation/EllipsisMenu/index.tsx (hunk 1 above)
Comment (fernandomg) [reply src/navigation/EllipsisMenu/index.tsx:54]: something around...

```typescript
  const handleClick: React.MouseEventHandler<'onclick'> = (event) => setAnchor(event.currentTarget);
```

seems to work.

Probably too specific, but in this case, I think it's enough.

Comment (nicosampler) [reply src/navigation/EllipsisMenu/index.tsx:54]: does not work for me

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

File: src/ethereum/AddressInfo/addressinfo.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+export const Address = (): React.ReactElement => (
+  <AddressInfo address={address} />
+);
+
+export const WithShortAddress = (): React.ReactElement => (
+  <AddressInfo address={address} shortenAddress={4} />
+);
+
+export const WithOwner = (): React.ReactElement => (
```
Comment (mikheevm) [inline src/ethereum/AddressInfo/addressinfo.stories.tsx:24]: WIth a name, not necessarily an owner

File: src/ethereum/AddressInfo/addressinfo.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+export const WithIdenticon = (): React.ReactElement => (
+  <AddressInfo address={address} showIdenticon shortenAddress={4} />
+);
+
+export const WithButtons = (): React.ReactElement => (
+  <AddressInfo
+    address={address}
+    name="Owner 1"
+    showIdenticon
+    showCopy
```
Comment (mikheevm) [inline src/ethereum/AddressInfo/addressinfo.stories.tsx:37]: showCopyWhat?

File: src/ethereum/AddressInfo/addressinfo.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    showEtherscan
```
Comment (mikheevm) [inline src/ethereum/AddressInfo/addressinfo.stories.tsx:38]: showEtherscanWhat?

File: src/ethereum/AddressInfo/index.tsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+  shortenAddress?: number;
+  className?: string;
+  name?: string;
+  showIdenticon?: boolean;
+  showCopy?: boolean;
+  menuItems?: EllipsisMenuItem[];
+  showEtherscan?: boolean;
+};
+
+const getShortAddress = (text: string, shortenAddress: number) =>
```
Comment (mikheevm) [inline src/ethereum/AddressInfo/index.tsx:49]: I think this wrapper over wrapper is unnecessary, you could use textShortener just fine

File: src/navigation/EllipsisMenu/index.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    componentSubtitle: 'Ellipsis Menu component used in Safe Multisig.',
+  },
+};
+
+const items: EllipsisMenuItem[] = [
+  { label: 'Item 1', disabled: true, onClick: console.log },
+  { label: 'Item 2', onClick: console.log },
+];
+
+export const Loader = (): React.ReactElement => (
```
Comment (mikheevm) [inline src/navigation/EllipsisMenu/index.stories.tsx:18]: Why is this called Loader?

File: src/navigation/EllipsisMenu/index.tsx [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+import { FixedIcon } from '../..';
+
+const MenuWrapper = styled.div`
+  display: flex;
+`;
+
+const MenuItemWrapper = styled.div`
+  :focus {
+    outline: none;
+  }
```
Comment (mikheevm) [inline src/navigation/EllipsisMenu/index.tsx:16]: http://www.outlinenone.com/

File: src/navigation/EllipsisMenu/index.tsx [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        </IconWrapper>
+        <Menu
+          anchorEl={anchorEl}
+          id="simple-menu"
+          keepMounted
+          onClose={closeMenuHandler}
+          open={Boolean(anchorEl)}>
+          {menuItems.map((i, index) => {
+            return (
+              <MenuItemWrapper key={index}>
```
Comment (mikheevm) [inline src/navigation/EllipsisMenu/index.tsx:79]: you could use label as a key

File: src/navigation/EllipsisMenu/index.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        <IconWrapper onClick={handleClick} onKeyDown={handleClick}>
+          <FixedIcon type="options" />
+        </IconWrapper>
+        <Menu
+          anchorEl={anchorEl}
+          id="simple-menu"
+          keepMounted
+          onClose={closeMenuHandler}
+          open={Boolean(anchorEl)}>
+          {menuItems.map((i, index) => {
```
Comment (mikheevm) [inline src/navigation/EllipsisMenu/index.tsx:77]: i -> `item`

File: src/navigation/EllipsisMenu/index.tsx (hunk 1 above)
Comment (mikheevm) [reply src/navigation/EllipsisMenu/index.tsx]: Screenshot from 2020-07-29 14 59 24

Material UI docs have both JS/TS examples with types :wink:

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

File: src/navigation/EllipsisMenu/index.tsx (hunk 1 above)
Comment (nicosampler) [reply src/navigation/EllipsisMenu/index.tsx]: I know and I already test it. But didn't work.

image

 Could you try yourself? Perhaps it's a problem with my computer.

Comment (mikheevm) [reply src/navigation/EllipsisMenu/index.tsx]: If you read through error message it says that it cannot assign to type of null (which is the inferred type of anchorEl). I already pushed a fix

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by fernandomg** [comment]
shouldn't this be closing #23 instead of #40?

@alongoni, @nicosampler

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

File: src/inputs/ButtonLink/index.tsx [hunk 10]
```diff
@@ -29,12 +30,13 @@ const StyledButtonLink = styled.button<Props>`
 const ButtonLik = ({
```
Comment (mikheevm) [inline src/inputs/ButtonLink/index.tsx:30]: typo

File: src/navigation/EllipsisMenu/index.tsx [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+      <MenuWrapper>
+        <IconWrapper onClick={handleClick}>
+          <FixedIcon type="options" />
+        </IconWrapper>
+        <StyledMenu
+          anchorEl={anchorEl}
+          keepMounted
+          onClose={closeMenuHandler}
+          open={Boolean(anchorEl)}>
+          {menuItems.map((item, i) => (
```
Comment (mikheevm) [inline src/navigation/EllipsisMenu/index.tsx:85]: i is unused

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr26--safereactcomponents.review.gnosisdev.com

**Pull request merged**